datamade/openness-project-nmid
# Add ability to redact name and address from donor records

## Overview

See title. Redacting information from one contribution record will also redact other contributions from the same donor.

Only authenticated users can redact records.

Redacted records should appear as redacted on candidate and committee detail pages, as well as in contribution exports. They should also be excluded from site search results.

### Demo

<img width="1624" alt="Screenshot 2023-12-13 at 4 06 50 PM" src="https://github.com/datamade/openness-project-nmid/assets/12176173/6bd927a6-cebf-4707-b88b-043507573b89">
<img width="1624" alt="Screenshot 2023-12-13 at 4 06 43 PM" src="https://github.com/datamade/openness-project-nmid/assets/12176173/c2212047-6bc3-4b72-9e39-d01ef3e142ee">
<img width="1624" alt="Screenshot 2023-12-13 at 3 51 12 PM" src="https://github.com/datamade/openness-project-nmid/assets/12176173/13ec0395-fc8f-4740-b147-45e3cc6ad3f0">

## Testing Instructions

* View a contribution and confirm you don't see the redaction form.
* Create a super user on the deploy preview and log in.
* Navigate to a contribution and confirm you can see the redaction form.
* Redact the record.
* Navigate to the detail page of the recipient, and confirm the donor's details are redacted.
* Unredact the record.
* Navigate to the detail page of the recipient, and confirm the donor's detail are not redacted.
